frontend/src/components/common/Resource/ScaleButton.tsx

```diff
@@ -127,11 +127,11 @@ function ScaleDialog(props: ScaleDialogProps) {
   const dispatchHeadlampEvent = useEventCallback(HeadlampEventType.SCALE_RESOURCE);
 
   function getNumReplicas() {
-    if (!resource?.spec) {
+    if (!('spec' in resource)) {
       return -1;
     }
 
-    return parseInt(resource.spec.replicas);
+    return parseInt((resource as any).spec.replicas);
```

Comment:
The scale dialog is only applied to some types of resources, maybe we should set those in the declaration of the resource in the dialog props' interface, rather than adding this bypass here.

Reply:
done